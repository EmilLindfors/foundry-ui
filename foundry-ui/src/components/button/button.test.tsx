import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Button from ".";
import { ThemeProvider } from "emotion-theming";

describe("<Button /> tests", () => {

  
  it("should render as default button", () => {

    const theme = {
      colors: {
        primary: "hotpink"
      }
    };

    const {container} = render(
      <ThemeProvider theme={theme}>
    <Button onClick={() => {}}>test</Button>
    </ThemeProvider>);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render a primary and secondary button", () => {
    // Arrange
    const variant = "primary";
    const variant2 = "secondary";
    const theme = {
      colors: {
        primary: "hotpink",
        secondary: "tomato"
      }
    };

    // Act
    const tree = render(
      <ThemeProvider theme={theme}>
        <Button color={variant} onClick={() => {}}>
          test
        </Button>
        <Button color={variant2} onClick={() => {}}>
          test
        </Button>
      </ThemeProvider>
    );

    // Assert
    expect(tree.container).toMatchSnapshot();
 
  });
  it("should call the click callback handler", () => {
    // Arrange
    const content = "Hello, World!";
    const onClick = jest.fn();
    const theme = {
      colors: {
        primary: "hotpink"
      }
    };
    // Act
    const { getByText } = render(
      <ThemeProvider theme={theme}>
    <Button onClick={onClick}>{content}</Button>
    </ThemeProvider>
    );
    fireEvent.click(getByText(content));

    // Assert
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
