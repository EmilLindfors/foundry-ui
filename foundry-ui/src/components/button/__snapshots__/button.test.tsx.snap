// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests should render a primary and secondary button 1`] = `
<div>
  <button
    class="css-vc6egq-Button"
  >
    test
  </button>
  <button
    class="css-1doieix-Button"
  >
    test
  </button>
</div>
`;

exports[`<Button /> tests should render as default button 1`] = `
<button
  class="css-vc6egq-Button"
>
  test
</button>
`;
